model/pod/pod.go

```diff
@@ -80,6 +80,7 @@ func (o *TaskIntentPodOptions) Validate(ecsConf evergreen.ECSConfig) error {
 	if o.OS == OSWindows {
 		catcher.Wrap(o.WindowsVersion.Validate(), "must specify a valid Windows version")
 	}
+	catcher.NewWhen(!utility.StringSliceContains(ecsConf.AllowedImages, o.Image), "image not allowed")
```

Comment:
@hadjri Just a heads up that this will cause the current image to fail unless it's set in staging/production.

Reply:
I set it in staging for "hadjri/evg-container-self-testing", so unless there's another one actively being used, we should be okay.